### PR TITLE
feat(database): size

### PR DESCRIPTION
**Pull request by alexirion10**
return database size
return server remaining disk size

tested on a few of our inhouse / QA boxes

### CODE REVIEW

File: src/SmoFacade/Server.cs [hunk 1]
```diff
@@ -57,6 +57,34 @@ public void DropLogin(LoginProperties login)
       _server.Logins[login.Name].DropIfExists();
     }
 
+    public decimal DatabaseSizeMb(string dbName)
+    {
+      string query = "SELECT total_size_mb = (SUM(size) * 8. / 1024) " + 
+                     "FROM sys.master_files WITH(NOWAIT) " +
```
Comment (Kaneraz) [inline src/SmoFacade/Server.cs:63]: Nice, this includes log size? That's a good metric to use.

File: src/SmoFacade/Server.cs [hunk 2]
```diff
@@ -57,6 +57,34 @@ public void DropLogin(LoginProperties login)
...
+      reader.Read();
+      return (decimal)reader["total_size_mb"];
+    }
+
+    public int RemainingDiskMb()
+    {
+      string query = "SELECT DISTINCT CONVERT(INT, dovs.available_bytes/1048576.0) AS free_size_mb " +
+                     "FROM sys.master_files mf " +
+                     "CROSS APPLY sys.dm_os_volume_stats(mf.database_id, mf.FILE_ID) dovs " +
+                     "WHERE dovs.logical_volume_name = 'Data'";
```
Comment (Kaneraz) [inline src/SmoFacade/Server.cs:79]: Is logical_volume_name always data? or is that just convention?

File: src/SmoFacade/Server.cs (hunk 1 above)
Comment (alexirion10) [reply src/SmoFacade/Server.cs:63]: correct, the `SUM()` aggregate on the `group by` combines log and data size

File: src/SmoFacade/Server.cs (hunk 2 above)
Comment (alexirion10) [reply src/SmoFacade/Server.cs:79]: Not sure what you're asking here.  Other values for `logical_volume_name` include TempDB, which is mounted on the T:\ volume in our case.
The query above aggregates everything on the D:\ drive which includes log and data files

Comment (alexirion10) [reply src/SmoFacade/Server.cs]: discussed with Kaneraz offline, updated query to match on SERVERPROPERTY('InstanceDefaultDataPath')

File: src/SmoFacade/Server.cs [hunk 3]
```diff
@@ -59,10 +59,10 @@ public void DropLogin(LoginProperties login)
 
     public decimal DatabaseSizeMb(string dbName)
     {
-      string query = "SELECT total_size_mb = (SUM(size) * 8. / 1024) " + 
-                     "FROM sys.master_files WITH(NOWAIT) " +
-                     $"WHERE database_id = DB_ID('{dbName}') " + 
-                     "GROUP BY database_id";
+      var query = "SELECT total_size_mb = (SUM(size) * 8. / 1024) " +
+                  "FROM sys.master_files WITH(NOWAIT) " +
+                  $"WHERE database_id = DB_ID('{dbName}') " +
```
Comment (Kaneraz) [inline src/SmoFacade/Server.cs:64]: Can we parameterize dbName since it's a string value?

**Review by Kaneraz — CHANGES_REQUESTED** [review]

File: src/SmoFacade/Server.cs (hunk 3 above)
Comment (alexirion10) [reply src/SmoFacade/Server.cs:64]: yep good point will change that

**Review by Kaneraz — APPROVED** [review]
Thanks, it looks good.

**Comment by alexirion10** [comment]
closes https://github.com/factset/AgDatabaseMove/issues/46

**Pull request merged**